src/invert/parderiv/impls/cyclic/cyclic.cxx

```diff
@@ -120,20 +130,25 @@ const Field3D InvertParCR::solve(const Field3D &f) {
       rfft(alignedField(x, y + localmesh->ystart), localmesh->LocalNz, &rhs(y + y0, 0));
 
     // Set up tridiagonal system
-    for(int k=0; k<nsys; k++) {
-      BoutReal kwave=k*2.0*PI/coord->zlength(); // wave number is 1/[rad]
+    for (int k = 0; k < nsys; k++) {
+      BoutReal kwave=k*2.0*PI/coord->zlength(); // wave number is 1/length
       for (int y = 0; y < localmesh->LocalNy - 2 * localmesh->ystart; y++) {
 
         BoutReal acoef = A(x, y + localmesh->ystart); // Constant
         BoutReal bcoef =
             B(x, y + localmesh->ystart) / coord->g_22(x, y + localmesh->ystart); // d2dy2
         BoutReal ccoef = C(x, y + localmesh->ystart);                            // d2dydz
         BoutReal dcoef = D(x, y + localmesh->ystart);                            // d2dz2
-        BoutReal ecoef = E(x, y + localmesh->ystart);                            // ddy
+        BoutReal ecoef =
+            E(x, y + localmesh->ystart)
+            + sg(x, y + localmesh->ystart)*B(x, y + localmesh->ystart);          // ddy
+
+        if (coord->non_uniform) {
+          ecoef += bcoef * coord->d1_dy(x, y + localmesh->ystart);
+        }
 
         bcoef /= SQ(coord->dy(x, y + localmesh->ystart));
-        ccoef /= coord->dy(x, y + localmesh->ystart) * coord->dz;
-        dcoef /= SQ(coord->dz);
```

Comment:
Where did this line with dcoef go?

Reply:
The `dz` factor is already included in the definition of `kwave` used here, which uses `zlength`

Reply:
Okay, thanks.